### PR TITLE
feature: Update cases in [Incorrect nesting of HTML tags]

### PR DESCRIPTION
**Pull request by hydRAnger**
It took me much time to debug this issue during my development process, and since the error logs(you can see below) given are not very clear, I hope it will help others who encounter the same problem.

image

ref: 
- [Are you allowed to nest a link inside of a link?](https://stackoverflow.com/questions/9882916/are-you-allowed-to-nest-a-link-inside-of-a-link)
- [Interactive Content](https://html.spec.whatwg.org/#interactive-content-2)
image

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 6c52279699c1bd961ed86bc241b1fac4c16c66cf

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 6c52279699c1bd961ed86bc241b1fac4c16c66cf

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: errors/react-hydration-error.mdx [hunk 1]
```diff
@@ -15,6 +15,7 @@ Hydration errors can occur from:
 1. Incorrect nesting of HTML tags
    1. `<p>` nested in another `<p>` tag
    2. `<div>` nested in a `<p>` tag
+   3. `<a>/<button>/etc.` nested in another `<a>/<button>/etc.` tag ([Interactive Content includes anchor tags](https://html.spec.whatwg.org/#interactive-content-2))
```
Comment (styfle) [inline errors/react-hydration-error.mdx:18]:
```suggestion
   3. [Interactive Content](https://html.spec.whatwg.org/#interactive-content-2) cannot be nested (`<a> nested in a `<a>` tag, `<button>` nested in a `<button>` tag, etc)
```

File: errors/react-hydration-error.mdx [hunk 2]
```diff
@@ -15,6 +15,7 @@ Hydration errors can occur from:
 1. Incorrect nesting of HTML tags
    1. `<p>` nested in another `<p>` tag
    2. `<div>` nested in a `<p>` tag
+   3. [Interactive Content](https://html.spec.whatwg.org/#interactive-content-2) cannot be nested (`<a> nested in a `<a>` tag, `<button>` nested in a `<button>` tag, etc)
```
Comment (styfle) [inline errors/react-hydration-error.mdx:18]:
```suggestion
   3. [Interactive Content](https://html.spec.whatwg.org/#interactive-content-2) cannot be nested (`<a>` nested in a `<a>` tag, `<button>` nested in a `<button>` tag, etc)
```

**Review by styfle — APPROVED** [review]
Thanks!

**Pull request merged**